sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/PrivateEndpointConnectionTests.cs

```diff
@@ -129,17 +128,31 @@ protected async Task<PrivateEndpointResource> CreatePrivateEndpoint()
             vnet.AddressPrefixes.Add("10.0.0.0/16");
             vnet.DhcpOptionsDnsServers.Add("10.1.1.1");
             vnet.DhcpOptionsDnsServers.Add("10.1.2.4");
-            VirtualNetworkResource virtualNetwork = (await _resourceGroup.GetVirtualNetworks().CreateOrUpdateAsync(WaitUntil.Completed, vnetName, vnet)).Value;
-
-            var name = Recording.GenerateAssetName("pe-");
+            //VirtualNetworkResource virtualNetwork = (await _resourceGroup.GetVirtualNetworks().CreateOrUpdateAsync(WaitUntil.Completed, vnetName, vnet)).Value;
+            ResourceIdentifier subnetID;
+            if (Mode == RecordedTestMode.Playback)
+            {
+                subnetID = SubnetResource.CreateResourceIdentifier(_resourceGroup.Id.SubscriptionId, _resourceGroup.Id.Name, vnetName, "default");
+            }
+            else
+            {
+                using (Recording.DisableRecording())
+                {
+                    VirtualNetworkResource vnetResource = (await _resourceGroup.GetVirtualNetworks().CreateOrUpdateAsync(WaitUntil.Completed, vnetName, vnet)).Value;
+                    var subnetCollection = vnetResource.GetSubnets();
+                    //SubnetResource subnetResource = (await subnetCollection.CreateOrUpdateAsync(WaitUntil.Completed, subnetName2, subnetData)).Value;
+                    subnetID = vnetResource.Data.Subnets[0].Id;
+                }
+            };
+            //var name = Recording.GenerateAssetName("pe-");
```

Comment:
Same comment here, could we extract this method and put it into the test base class?

Reply:
fixed